plug.vim

```diff
@@ -2001,6 +2044,10 @@ function! s:shellesc_ps1(arg)
   return "'".substitute(escape(a:arg, '\"'), "'", "''", 'g')."'"
 endfunction
 
+function! s:shellesc_sh(arg)
+  return "'".substitute(a:arg, "'", "'\\\\''", 'g')."'"
```

Comment:
Pipeline passes and I didn't encounter any issues when running the tests locally or using my config so I assume this is good enough.